src/Microsoft.DotNet.Wpf/src/Shared/System/Windows/InterOp/OSVersionHelper.cs

```diff
@@ -139,6 +143,10 @@ static OSVersionHelper()
         [return: MarshalAs(UnmanagedType.I1)]
         static extern bool IsWindows10RS5OrGreater();
 
+        [DllImport(DllImport.PresentationNative, CallingConvention = CallingConvention.Cdecl)]
+        [return: MarshalAs(UnmanagedType.I1)]
+        static extern bool IsWindows10RS4OrGreater();
```

Comment:
Thanks, @ThomasGoulet73.  IsWindows10RS4OrGreater and the others are implemented in PresentationNative, which is in a separate internal WPF repo.

Reply:
Thank you @ryalanms! I didn't know that the PresentationNative is closed source and that the project in this repo is simply a redist. Is it planned to open-source PresentationNative or is the status still the same as #2554 ?

Reply:
There is a historical reason why IsWindows10RS4OrGreater etc are sitting in PresentationNative. It should be possible to move these to an open source assembly (esp to minimize the footprint of closed-source where possible) without problems. 

When we originally added these years ago (circa .NET 4.6.1 probably), the only way to do reliable OS version checks was by calling RtlVerifyVersionInfo API’s - VerifyVersionInfo didn’t work well and it required application manifests which was not feasible from a DLL. RtlVerifyVersionInfo at that time was only documented as being available via  NtosKrnl.lib - the ntdll.dll export was well known but hadn’t been officially documented at that time (on MSDN/docs.Microsoft.com). So instead of just P/Invoking into RtlVerifyVersionInfo, we linked PresentationNative to NtosKrnl.lib to get access to RtlVerifyVersionInfo and exported our own functions from there to P/Invoke into (remember, .NET and thus WPF only uses documented API’s). 

Forward a few years  RtlVerifyVersionInfo was documented as a dual mode API exported out of ntdll.dll (and NtosKrnl.exe) and this complicated way of consuming taht API was no longer needed (now instead of linking to NtosKrnl.lib, we could P/Invoke into ntdll.dll if that’s what we preferred) - but we never changed the approach.